tests/checkpointing/test_model_checkpoint.py

```diff
@@ -1020,3 +1021,35 @@ def __init__(self, hparams):
     else:
         # make sure it's not AttributeDict
         assert type(ckpt[model.CHECKPOINT_HYPER_PARAMS_KEY]) == hparams_type
+
+
+def test_model_checkpoint_with_training_epoch_end(tmpdir):
+
+    """
+    This test assert ModelCheckpoint a warming is issued when monitor metric is used in training_epoch_end
+    """
+    class TestedModel(BoringModel):
+
+        def training_step(self, batch, batch_idx):
+            output = self.layer(batch)
+            loss = self.loss(batch, output)
+            self.log('train_loss', loss)
+            return {"loss": loss}
+
```

Comment:
Why do you need to override `training_step`? BoringModel's is the same

```suggestion
```

Reply:
I just adds self.log to training_step and will be used to compute coverage.

Reply:
We need to add `self.log` to training_step, so coverage gets computed on this new added piece of code.